### PR TITLE
Admins can create and edit events with tags, venue, date and time

### LINKED ISSUE #270: Auth: As an owner or chapter admin I can create events and set their venue type, venue, video url, tags and capacity.
Depends on https://github.com/freeCodeCamp/chapter/pull/814

The UI is looking solid, so the main outstanding issue is to add auth and test it, similarly to 
https://github.com/freeCodeCamp/chapter/issues/273

- [x] As an owner I can create events in any Chapter
- [x] As an admin of Chapters, I can create events in those Chapters (but no others)
- [x] As any other role, I cannot create events

The main thing to check is that requests from unauthorised users fail, but it's also helpful to confirm that the UI reflects their role.

### PR DESCRIPTION
**Pull request by tomiiide**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `master` branch of Chapter.

Closes #270 

Admins can create and edit events with tags, venues, capacity, date, and time.
Added a loader to the form buttons.

### CODE REVIEW

File: client/components/Dashboard/Events/EventForm.tsx [hunk 1]
```diff
@@ -91,7 +91,9 @@ const EventForm: React.FC<IEventFormProps> = ({
             labelId="venue-label"
             required
             value={venueId}
-            onChange={e => setVenueId(e.target.value as number)}
+            onChange={e => {
```
Comment (Zeko369) [inline client/components/Dashboard/Events/EventForm.tsx:94]: Do you have a specific reason on why to do this?

File: client/components/Dashboard/Events/EventForm.tsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ const EventForm: React.FC<IEventFormProps> = ({
   const { control, handleSubmit } = useForm();
   const styles = useFormStyles();
 
-  const [venueId, setVenueId] = useState<number | undefined>(0);
+  const [venueId, setVenueId] = useState<number | undefined>(1);
```
Comment (Zeko369) [inline client/components/Dashboard/Events/EventForm.tsx:52]: Maybe better to use -1 to keep it unselected by default

**Comment by allella** [comment]
@tomiiide  we're getting momentum again on the project.  Would you have time to readdress Fran's questions and the branch conflicts on this PR?

Thanks

**Comment by ojeytonwilliams** [comment]
I'm closing this for now since there hasn't been any activity for a while

If you have time to update this, please go ahead and reopen this.

**Pull request closed without merging**